app/models/outboundLink/OutboundLink.scala

```diff
@@ -16,182 +16,217 @@
+      case _              => controllers.goodsRecord.routes.CommodityCodeController.onSubmitCreate(mode).url
+    }
   }
 
   // Longer Commodity Code Page
-  case object FindLongCommodity extends OutboundLink {
-    val link: String        = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
-    val linkTextKey: String = "longerCommodityCode.linkText"
+  case class FindLongCommodity(mode: Mode, recordId: String) extends OutboundLink {
+    val link: String            = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
+    val linkTextKey: String     = "longerCommodityCode.linkText"
+    val originatingPage: String =
+      controllers.categorisation.routes.LongerCommodityCodeController.onPageLoad(mode, recordId).url
   }
 
   // Has Correct Goods Page
-  case object FindCommodityHasCorrectGoods extends OutboundLink {
-    val link: String        = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
-    val linkTextKey: String = "hasCorrectGoods.p2.linkText"
+  case class FindCommodityHasCorrectGoods(mode: Mode, recordId: Option[String]) extends OutboundLink {
+    val link: String            = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
+    val linkTextKey: String     = "hasCorrectGoods.p2.linkText"
+    val originatingPage: String = recordId match {
+      case Some(recordId) => controllers.routes.HasCorrectGoodsController.onSubmitUpdate(mode, recordId).url
+      case _              => controllers.routes.HasCorrectGoodsController.onSubmitCreate(mode).url
+    }
   }
 
   // Assessment Page
-  case object FindCommodityAssessments extends OutboundLink {
-    val link: String        = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
-    val linkTextKey: String = "assessment.linkText"
+  case class FindCommodityAssessments(mode: Mode, recordId: String, assessmentNumber: Int, isReassessment: Boolean)
+      extends OutboundLink {
+    val link: String            = "https://www.trade-tariff.service.gov.uk/xi/find_commodity"
+    val linkTextKey: String     = "assessment.linkText"
+    val originatingPage: String = if (isReassessment) {
+      controllers.categorisation.routes.AssessmentController
+        .onPageLoadReassessment(mode, recordId, assessmentNumber)
+        .url
+    } else {
+      controllers.categorisation.routes.AssessmentController.onPageLoad(NormalMode, recordId, assessmentNumber).url
+    }
+  }
+
+  case class AssessmentDynamicLink(
+    link: String,
+    mode: Mode,
+    recordId: String,
+    assessmentNumber: Int,
+    isReassessment: Boolean
+  ) extends OutboundLink {
+    val linkTextKey: String     = "assessment.regulationUrl.linkText"
+    val originatingPage: String = if (isReassessment) {
+      controllers.categorisation.routes.AssessmentController
+        .onPageLoadReassessment(mode, recordId, assessmentNumber)
+        .url
+    } else {
+      controllers.categorisation.routes.AssessmentController.onPageLoad(mode, recordId, assessmentNumber).url
+    }
```

Comment:
This val originatingPage looks the same as the one above. Could it be extracted? The only difference I see is the 'mode' parameter in the else - should it not be NormalMode in  both

Reply:
I had meant the above case class For my comment I meant the above case class FindCommodityAssessments, they both pass in mode. Looked like repeated code. Happy to approve